src/components/pages/SingleMoviePage.vue

```diff
@@ -0,0 +1,129 @@
+<script>
+import { defineComponent } from 'vue';
+import BreadCrumbs from '@/components/common/BreadCrumbs.vue';
+import SectionTitle from '@/components/common/Section/SectionTitle.vue';
+import AppTag from '@/components/common/App/AppTag.vue';
+import MovieLengthOrYear from '@/components/common/Movie/MovieLengthOrYear.vue';
+import { getMovieById } from '@/api/service/Movies';
+import ScreeningsSection from '@/components/views/ScreeningsSection.vue';
+import TheContainer from '@/components/common/TheContainer.vue';
+
+export default defineComponent({
+  components: { BreadCrumbs, SectionTitle, AppTag, ScreeningsSection, MovieLengthOrYear, TheContainer },
+  data() {
+    return {
+      isRequestFinished: false,
+      movieData: {},
+      screenings: [],
+    }
+  },
+  computed: {
+    movieIdFromUrl() {
+      const url = window.location.pathname
+      const segments = url.split("/")
+      return segments[2]
+    },
```

Comment:
While this does work somewhat (it won't work properly in a few peculiar cases though 👀) - we can do better.

We can define a prop and... have our router set it up for us 💡 

For example, if we would have a route like:
```js
{
  name: 'SingleItemPage',
  path: '/items/:itemId',
  component: () => import('@/views/SingleItemPage.vue')
}
```

All we need to do is add `props: true` option to the route - then, the vue router will try to bind the value of `itemId` to the `SingleItemPage` component. All we need to do is define a prop for that on it and... that's it!
No more weird tinkering with window's location or checking on current route.

More on this:
https://router.vuejs.org/guide/essentials/passing-props.html#object-mode

Reply:
Wow, this is just brilliant! ✨ Thanks a lot for this hint! 😃